docs/DeveloperGuide.md

```diff
@@ -1,13 +1,71 @@
 # Developer Guide
 
+- [Acknowledgements](#acknowledgements)
+- [Design](#design)
+  - [Architecture](#architecture)
+- [Implementation](#implementation)
+  - [`add` feature](#add-feature)
+  - [`delete` feature](#delete-feature)
+  - [`set` spending limit feature](#set-spending-limit-feature)
+  - [`list` feature](#list-feature)
+  - [`help` feature](#help-feature)
+  - [[Proposed] `find` feature](#proposed-find-feature)
+- [Product scope](#product-scope)
+  - [Target user profile](#target-user-profile)
+  - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Non-functional requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for manual testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
-
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
+## Design
+### Architecture
+![](Architecture Diagram.jpg)
+
+Fig 1 - Architecture Diagram
+
+The Architecture Diagram given above explains the high-level design of the App.
+
+**Main components of the architecture:**
+- `Main Spendvelope`: Initializes and makes use of the services in each component.
+- `TextUi`: Interacts with the user.
+- `Parser`: Makes sense of user commands.
+- `Commands`: The collection of executable commands on the App.
+- `Storage`: Reads data from, and writes data to the hard disk.
+- `LimitManager`: Allows the user to set an expenditure limit. Holds this limit in memory.
+- `RecordManager`: Allows user to add, delete, view records. Holds this list in memory.
+- `ExpenseManager`: Tracks the total expenditure from all records.
+
+## Implementation
+### `add` feature
+When command `add` is called, a `Record` object is created and added to the record list in `RecordManager`. Depending 
+on the input, either a `Product` or a `Subscription` object will be created. They extend `Record` with their own 
+properties. `ExpenseManager` will add the price of the new record to total expense using `addToExpense(price)`.
+
+### `delete` feature
+When command `delete` is called, the corresponding record will be deleted by `RecordManager` from the record list. 
```

Comment:
No sequence diagram, but more importantly there's no example command and data. Hence no way to visualise what you're trying to explain